OpenRA.Game/Graphics/RgbaColorRenderer.cs

```diff
@@ -178,18 +196,12 @@ void DrawConnectedLine(float2[] points, float width, Color color, bool closed)
 			}
 		}
 
-		public void DrawLine(float2[] points, float width, Color color, bool connectSegments = false)
+		public void DrawLine(IEnumerable<float2> points, float width, Color color, bool connectSegments = false)
 		{
 			if (!connectSegments)
-			{
-				if (points.Length < 2)
-					return;
-
-				for (var i = 1; i < points.Length; i++)
-					DrawLine(points[i - 1], points[i], width, color);
-			}
+				DrawDisconnectedLine(points, width, color);
 			else
-				DrawConnectedLine(points, width, color, false);
+				DrawConnectedLine(points as float2[] ?? points.ToArray(), width, color, false);
```

Comment:
Performance optimization if somebody sent an array in the first place. If they did, we just cast rather than having to create a second copy of the array.